### DISCUSSION TITLE
The value \"4294967295\" is invalid for option \"family\" when deploy to my server

### DISCUSSION THREAD

**Discussion by arsaphone** [fastify/fastify#2478] [General] [answered]
hello,

before i created this duscussion i need to define first that
SV = server
LP= local pc (my laptop)

so it wont be confussing. hehe.

i created an API with fastify 
to do some CRUD to mysql database. all sets and good on my docker desktop on LP with the same network.

and this is my app.js

```js
const fastify = require('fastify');
const isDocker = require('is-docker');
const cors = require('cors');
const {
  APP_PORT
} = require('./environment');

const app = fastify({
  bodyLimit: 1048576 * 2,
  logger: {
    prettyPrint: true
  }
});

app.get('/', async (request, reply) => {
  return {
    hello: 'world'
  }
})

app.register(require('./routes/api'), {
  prefix: 'api'
});

const start = async () => {
  try {
    if (isDocker()) {
      await app.listen(APP_PORT, '0.0.0.0')
    } else {
      await app.listen(APP_PORT)
    }
    app.log.info(`server listening on ${app.server.address().port}`)
  } catch (err) {
    app.log.error(err)
    process.exit(1)
  }
}
start()

```
and this is my .env

```
// in your .env
NODE_ENV=development
/* API INFO */
TIME_ZONE=
APP_PORT=4004

// in you .env.* (.env.local)
/* DATABASE INFORMATION */
DB_NOSQL_HOST=
DB_NOSQL_USER=
DB_NOSQL_PASSWORD=
DB_NOSQL_NAME=
DB_NOSQL_PORT=27017
DB_SQL_CLIENT=mysql
DB_SQL_HOST=mysql_server
DB_SQL_USER=root
DB_SQL_PASSWORD=password
DB_SQL_NAME=mysqldb
DB_SQL_PORT=3306
```

and i push the image with the current image using on LP to SV.
i tried to push and set to my server (ubuntu 16.04)
and sets all the docker setting just same as on LP. same name, same network.

tried to ping from the fastify container to mysql-server and it got response
i found it confusing when DB_SQL_HOST uses string it will get the **The value \"4294967295\" is invalid for option \"family\"** error.

anyone have a solution maybe?

**Comment by mcollina** [comment]
I think that's a problem in the part of your code that we are not seeing.

**Reply by arsaphone** [reply]
i don't get it. why would it be different from docker on my laptop and server?

**Reply by mcollina** [reply]
You have some bugs in your setup/business logic that is setting some bad value for a `family` option of another module that is not part of Fastify. I think the problem is in how you load your environment variables and/or a value of those.

**Reply by arsaphone** [reply]
```
var mysql = require("mysql");

var con = mysql.createPool({
  connectionLimit: 5,
  host: process.env.DB_SQL_HOST,
  user: process.env.DB_SQL_USER,
  password: process.env.DB_SQL_PASSWORD,
  database: process.env.DB_SQL_NAME,
  port: process.env.DB_SQL_PORT,
});

module.exports = con;
```
you mean this file?

that's the mysqlconnection.js which pointing to .env

**Reply by arsaphone** [reply]
update :
its working on my docker local computer but still getting that error when deployed to server.

**Reply by kibertoad** [reply]
This has nothing to do with fastify, and everything with your environment configuration. I would suggest you to track where you are loading config value for family from, and check server config for that.

**Reply by arsaphone** [reply]
> 
> 
> This has nothing to do with fastify, and everything with your environment configuration. I would suggest you to track where you are loading config value for family from, and check server config for that.

i tried to clone and build the image with -t in linux server 
and it gave me the  **The value "4294967295" is invalid for option "family"**  error when do the step on dockerfile `RUN npm install`
should i track this one?
 i did it on windows and it worked perfectly tho.

edit : more explanation.

**Reply by jsumners** [reply, marked as answer]
https://duckduckgo.com/?q=The+value+"4294967295"+is+invalid+for+option+"family"&t=vivaldi&ia=web&iai=r1-0&page=1&adx=prdsdc&sexp={"v7exp"%3A"a"%2C"sltexp"%3A"b"%2C"prodexp"%3A"b"%2C"prdsdexp"%3A"c"%2C"wiadrk"%3A"b"}

The first search result. I literally just highlighted the bolded part of your last message and did a right click then "Search for ...".
